core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -112,12 +112,13 @@ class LogSegment(val log: FileMessageSet,
    * @param startOffset A lower bound on the first offset to include in the message set we read
    * @param maxSize The maximum number of bytes to include in the message set we read
    * @param maxOffset An optional maximum offset for the message set we read
+   * @param maxPosition An optional maximum position in the log segment that should be exposed for read.
```

Comment:
@junrao I thought it is optional because it has a default value. i.e. caller can choose not to specify the maxPosition. 
But I think we should probably cap the maxPosition even if maxOffset is provided. I will update the patch.